### PR TITLE
feat: add week 4 homework

### PR DESCRIPTION
**Pull request by khankaholic**
**Week 4 Homework for RAG Foundation**

authors: 
@cin-toy 
@cin-tadashi
@cin-kay

### CODE REVIEW

**Review by Tin-Hoang — DISMISSED** [review]
Approved.

File: rag-foundation/scripts/main.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        prompt_template = """Question: {}\n\nGiven context: {}\n\nAnswer:"""
+
+        rag_pipeline = RAGPipeline(vector_store, prompt_template=prompt_template)
+
+        for q in values["qas"]:
+            # for each question in the paper
+            query = q["question"]
+            question_ids.append(q["question_id"])
+
+            # NOTE: If you just want to retrieve the top 5 relevant documents
```
Comment (cin-toy) [inline rag-foundation/scripts/main.py]: Modify to: "If you just want to retrieve the top_k relevant documents..."

File: rag-foundation/predictions.jsonl
Comment (cin-toy) [inline rag-foundation/predictions.jsonl:1]: Should we leave the prediction file here? (just my concern...)

**Review by cin-toy — COMMENTED** [review]
Overall, it seems fine to me. Just 2 comments from me.

File: rag-foundation/predictions.jsonl
Comment (khankaholic) [reply rag-foundation/predictions.jsonl:1]: @cin-toy we can leave it as sample prediction, lemma rename it.

File: rag-foundation/scripts/main.py (hunk 1 above)
Comment (khankaholic) [reply rag-foundation/scripts/main.py]: fixed in [63fad35](https://github.com/Cinnamon/ai-bootcamp-2024/pull/2/commits/63fad35519d68e0436f079e2378440a149878e8b)

File: rag-foundation/predictions.jsonl
Comment (cin-toy) [reply rag-foundation/predictions.jsonl:1]: Thank you @cin-kay. I agree with you.

File: rag-foundation/scripts/main.py (hunk 1 above)
Comment (cin-toy) [reply rag-foundation/scripts/main.py]: Thank you @cin-kay.

File: rag-foundation/predictions.jsonl
Comment (khankaholic) [reply rag-foundation/predictions.jsonl:1]: resolved in this https://github.com/Cinnamon/ai-bootcamp-2024/pull/2/commits/dca8319b7be99fe81a0db627fe07b3e8abd818f5

**Review by cin-toy — APPROVED** [review]
Approval from me.

**Pull request merged**